core/src/main/java/io/aiven/klaw/helpers/db/rdbms/SelectDataJdbc.java

```diff
@@ -1476,16 +1486,26 @@ public Map<String, Map<String, Long>> getAclRequestsCounts(
       List<Object[]> aclRequestsStatusObj =
           aclRequestsRepo.findAllAclRequestsGroupByStatusMyTeam(teamId, tenantId);
       updateMap(statusCountsMap, aclRequestsStatusObj);
-    } else if (RequestMode.TO_APPROVE == requestMode) {
+    } else if (RequestMode.TO_APPROVE == requestMode || RequestMode.MY_APPROVALS == requestMode) {
       List<Object[]> aclRequestsOperationTypObj =
           aclRequestsRepo.findAllAclRequestsGroupByOperationTypeAssignedToTeam(teamId, tenantId);
       updateMap(operationTypeCountsMap, aclRequestsOperationTypObj);
 
       List<Object[]> aclRequestsStatusObj =
           aclRequestsRepo.findAllAclRequestsGroupByStatusAssignedToTeam(teamId, tenantId);
       updateMap(statusCountsMap, aclRequestsStatusObj);
-    }
+      if (RequestMode.MY_APPROVALS == requestMode) {
+        // Make sure to remove any requests which the requestor can not approve
+        List<Object[]> aclApprovalRequestsStatusObj =
```

Comment:
why not to replace with count query?
Count query is much faster since it doesn't need to load all the field values